// custom-react-accordion/dist/index.modern.js
import React, { createContext, useState, useContext } from 'react';
import PropTypes from 'prop-types';

var AccordionContext = createContext({
  active: 0,
  setActive: function setActive() {}
});

var AccordionWrapper = function AccordionWrapper(props) {
  var _useState = useState(0),
      active = _useState[0],
      setActive = _useState[1];

  return /*#__PURE__*/React.createElement(AccordionContext.Provider, {
    value: {
      active: active,
      setActive: setActive
    }
  }, /*#__PURE__*/React.createElement("div", {
    className: "accordion-wrapper"
  }, /*#__PURE__*/React.createElement("form", null, props.children)));
};

var AccordionItem = function AccordionItem(props) {
  var indexPlus;

  var indexCount = function indexCount(index) {
    indexPlus = index + 1;
    return indexPlus;
  };

  var _useContext = useContext(AccordionContext),
      active = _useContext.active,
      setActive = _useContext.setActive;

  var eventHandler = function eventHandler(e, index) {
    e.preventDefault();
    setActive(index);
  };

  return /*#__PURE__*/React.createElement("div", {
    className: "accordion-item"
  }, /*#__PURE__*/React.createElement("h3", {
    className: "accordion-title"
  }, /*#__PURE__*/React.createElement("button", {
    onClick: function onClick(e) {
      return eventHandler(e, props.index);
    },
    className: active === props.index ? 'active' : 'inactive',
    "aria-expanded": active === props.index ? 'true' : 'false',
    "aria-controls": 'sect-' + indexCount(props.index),
    "aria-disabled": active === props.index ? 'true' : 'false'
  }, /*#__PURE__*/React.createElement("span", {
    className: "title-wrapper"
  }, props.title), /*#__PURE__*/React.createElement("span", {
    className: "icon-wrapper"
  }, /*#__PURE__*/React.createElement("span", {
    className: active === props.index ? 'plus' : 'minus'
  })))), /*#__PURE__*/React.createElement("div", {
    className: "accordion-panel"
  }, /*#__PURE__*/React.createElement("div", {
    id: 'sect-' + indexCount(props.index),
    className: active === props.index ? 'panel-open' : 'panel-close'
  }, props.description)));
};

AccordionItem.propTypes = {
  index: PropTypes.number.isRequired,
  title: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired
};

export { AccordionItem, AccordionWrapper };
